### PR TITLE
store/copr: support EngineRoleWrite for disaggregated tiflash

### LINKED ISSUE #41978: support EngineRoleWrite for disaggregated tiflash
## Enhancement
For non-disaggregated tiflash mode, TiDB should dispatch MPPTask to tiflash whose label **contains** <engine, tiflash>, but **doesn't contain** <engine_role, write>. Because <engien_role, write> is just for disaggregated tiflash mode.

For disaggregated tiflash mode, TiDB shoudle dispatch MPPTask to tiflash whose label contains <engine, tiflash_compute>.

### PR DESCRIPTION
**Pull request by guo-shaoge**
### What problem does this PR solve?

Issue Number: close #41978

Problem Summary: As issue said.

### What is changed and how it works?
1. For non-disaggregated tiflash mode, should ignore tiflash whose label contains <engine_role, write>.
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -279,3 +279,5 @@ replace (
 	github.com/pingcap/tidb/parser => ./parser
 	go.opencensus.io => go.opencensus.io v0.23.1-0.20220331163232-052120675fac
 )
+
+replace github.com/tikv/client-go/v2 => /home/guojiangtao/work/client-go
```
Comment (guo-shaoge) [inline go.mod:283]: Wait: https://github.com/tikv/client-go/pull/728

**Review by XuHuaiyu — APPROVED** [review]

**Review by wshwsh12 — APPROVED** [review]

**Comment by guo-shaoge** [comment]
Wait for others pr to fix client-go.

**Pull request merged**